src/tram/tram/static/js/analyze.js

```diff
@@ -65,21 +123,21 @@ function renderMappings(sentence_id) {
     }
 
     sentence = stored_sentences[sentence_id];
-    mappings = sentence.mappings;
 
     $mappingTable = $(`<table id="mapping-table" class="table table-striped table-hover"><tbody></tbody></table>`);
-    var addButton = `<button class="btn btn-sm btn-success" data-toggle="modal" data-target="#addMappingModal">Add...</button>`;
+    var addButton = `<button class="btn btn-sm btn-success" data-bs-toggle="modal" data-bs-target="#addMappingModal">Add...</button>`;
 
     $mappingTable.append(`<tr><th>Technique ${addButton}</th><th>Confidence</th><th></th></tr>`);
-    for (i = 0; i < sentence.mappings.length; i++) {
+    for (let i = 0; i < sentence.mappings.length; i++) {
         var mapping = sentence.mappings[i];
         var $row = $(`<tr></tr>`);
         $row.append(`<td>${mapping.attack_id} - ${mapping.name}</td>`);
         $row.append(`<td>${mapping.confidence}%</td>`);
         $removeButton = $(`<button type="button" class="btn btn-sm btn-danger"><i class="fas fa-minus-circle"></i><`);
-        $removeButton.click( function() {
-            deleteMapping(sentence.id, mapping.id);
-        });
+        $removeButton.click(
+            // Need this callback to avoid closure issues when assigning onClick event
+            createCallback(sentence.id, mapping.id, false)
```

Comment:
Sorry for leading you astray on this one callback business. I forgot that JavaScript closures don't bind values in a loop unless you use `let`. I think changing `var mapping` to `let mapping` and then using arrow syntax would do the trick (might also need `let sentence`?), but I don't want to beat a dead horse. We can clean up the code in CTID-26.

Reply:
Good point, I was able to get this fixed after suggesting the `let` instead of `var`.